src/components/tools/ManipulatorTool.vue

```diff
@@ -38,38 +39,98 @@ export default defineComponent({
         : viewProxy.value.getInteractorStyle3D();
     });
 
-    const manipulator = props.manipulatorClass.newInstance(options.value);
+    const manipulators: any[] = [];
+    options.value.forEach((opt) =>
+      manipulators.push(props.manipulatorClass.newInstance(opt))
+    );
+
+    function updateStyle(
+      style: any,
+      newManipulators: any[],
+      oldManipulators: any[]
+    ) {
+      if (style !== null) {
+        oldManipulators.forEach((m) => {
+          if (m.isA('vtkCompositeMouseManipulator')) {
+            style.removeMouseManipulator(m);
+          }
+        });
+
+        newManipulators.forEach((m) => {
+          if (m.isA('vtkCompositeMouseManipulator')) {
+            style.addMouseManipulator(m);
+          }
+        });
+      }
+    }
 
+    /*
     watch(options, (newOptions) => {
-      if ('button' in newOptions) {
-        manipulator.setButton(newOptions.button);
+      // Remove all previous manipulators
+      while(manipulators.length > 0) {
+        const m = manipulators.pop();
+        intStyle.value.removeMouseManipulator(m);
+        m.delete();
       }
-      if ('shift' in newOptions) {
-        manipulator.setShift(newOptions.shift);
+
+      // Create all new manipulators
+      for(let i = 0; i < newOptions.length; ++i) {
+        manipulators.push(props.manipulatorClass.newInstance(newOptions[i]));
       }
-      if ('control' in newOptions) {
-        manipulator.setControl(newOptions.control);
+
+      updateStyle(intStyle.value, manipulators, []);
+    });
+    */
+
+    watch(options, (newOptions) => {
+      // Maintain the size of manipulators array equal to options array.
+      while (manipulators.length > newOptions.length) {
+        const m = manipulators.pop();
+        intStyle.value.removeMouseManipulator(m);
+        m.delete();
       }
+
+      const oldManipulators = Array.from(manipulators);
+
+      let i = manipulators.length;
+      while (manipulators.length < newOptions.length) {
```

Comment:
Can be simplified to `for (let i = manipulators.length; i < newOptions.length; i++)`